pkg/utils/utils.go

```diff
@@ -64,3 +76,13 @@ func ParseSubjectReference(subRef string) (common.Reference, error) {
 func TrimSpaceAndToLower(input string) string {
 	return strings.ToLower(strings.TrimSpace(input))
 }
+
+// ParseRequestKey parses key string to a structured RequestKey object.
+func ParseRequestKey(key string) RequestKey {
+	re := regexp.MustCompile(subjectPattern)
```

Comment:
What happens (ie. error message sent) if it doesn't match the regular expression?

Reply:
good point! This pattern should work for all strings including empty string, that's why I remove the length check and empty check.